core/services/vrf/vrf_simulate_blockchain_test.go

```diff
@@ -0,0 +1,153 @@
+package vrf_test
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/smartcontractkit/chainlink/core/adapters"
+	"github.com/smartcontractkit/chainlink/core/internal/cltest"
+	"github.com/smartcontractkit/chainlink/core/services/signatures/secp256k1"
+	"github.com/smartcontractkit/chainlink/core/services/vrf"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/core/store/models/vrfkey"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func registerExistingProvingKey(
+	t *testing.T,
+	coordinator coordinatorUniverse,
+	provingKey *vrfkey.PrivateKey,
+	jobID *models.ID,
+	vrfFee *big.Int,
+) {
+	var rawJobID [32]byte
+	copy(rawJobID[:], []byte(jobID.String()))
+	_, err := coordinator.rootContract.RegisterProvingKey(
+		coordinator.neil, vrfFee, coordinator.neil.From, pair(secp256k1.Coordinates(publicKey)), rawJobID)
+	require.NoError(t, err, "failed to register VRF proving key on VRFCoordinator contract")
+	coordinator.backend.Commit()
+}
+
+// TODO - this tests's eth client can be entirely replaced with the simulated backend once
+// the GethClient and RPCClient definitions are finished
+func TestIntegration_RandomnessRequest(t *testing.T) {
```

Comment:
Another nitpick, can you split this test into helper functions, maybe setup, assert, teardown as a first step. It's hard to figure out what the steps are. Also, can this single test be refactored into a suite, with a smaller number of assertions per unit?

Reply:
Another good suggestion.. I would maybe like to save that work for when we refactor the test to fully rely on the simulated backend - otherwise I think we'll end up just doing double the refactoring work. Does that seem reasonable? To your point though, I have always thought that a lot of our golang tests are incredibly verbose and difficult to follow.. and could share a lot more setup/teardown than then currently do.